src/robusta/core/model/base_params.py

```diff
@@ -70,6 +71,45 @@ def post_initialization(self):
     pass
 
 
+class ResourceInfo(BaseModel):
+    name: str
+    namespace: Optional[str]
+    kind: str
+    node: Optional[str]
+    container: Optional[str]
+
+
+class HolmesParams(ActionParams):
+
+    holmes_url: Optional[str]
+
+    @validator("holmes_url", allow_reuse=True)
+    def validate_protocol(cls, v):
+        if v and not v.startswith("http"):  # if the user configured url without http(s)
+            v = f"http://{v}"
+            logging.info(f"Adding protocol to holmes_url: {v}")
+        return v
+
+
+class AIInvestigateParams(HolmesParams):
+    """
+    :var resource: The resource related to this investigation. A resource has a `name` and `kind`, and may have `namespace` and `node`
+    :var investigation_type: The type of investigation: Issue/Service/Cluster/Custom
```

Comment:
this comment is a bit confusing since I see the code only sets it to `issue` or `analyze_problems` not Service/Cluster/Custom
Correct me if im wrong?

Reply:
good catch, fixed